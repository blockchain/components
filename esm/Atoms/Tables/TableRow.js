function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: flex-start;\n  align-items: center;\n  width: 100%;\n  padding: 30px 30px;\n  box-sizing: border-box;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
var BaseTableRow = styled.div(_templateObject());

var TableRow = function TableRow(_ref) {
  var children = _ref.children;
  return React.createElement(BaseTableRow, null, children);
};

TableRow.propTypes = {
  children: PropTypes.node
};
export default TableRow;