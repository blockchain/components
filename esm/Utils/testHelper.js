import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}