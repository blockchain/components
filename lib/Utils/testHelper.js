"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}