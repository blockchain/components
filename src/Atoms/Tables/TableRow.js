import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  padding: 30px 30px;
  box-sizing: border-box;
`

const TableRow = ({ children }) => <Wrapper>{children}</Wrapper>

TableRow.propTypes = {
  children: PropTypes.node,
}

export default TableRow
