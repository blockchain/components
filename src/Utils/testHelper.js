import { mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components'
import theme from '../Tools/theme'

export const mountWithTheme = (children) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
