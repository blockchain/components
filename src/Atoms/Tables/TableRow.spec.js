import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TableRow from './TableRow'

describe('TableRow', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TableRow>MyTableRow</TableRow>)

    expect(tree.find(TableRow)).toMatchSnapshot()
  })
})
