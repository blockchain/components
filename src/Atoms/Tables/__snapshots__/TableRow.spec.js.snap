// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1.875rem;
  box-sizing: border-box;
}

<TableRow>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  padding: 1.875rem;
  box-sizing: border-box;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        MyTableRow
      </div>
    </StyledComponent>
  </styled.div>
</TableRow>
`;
